src/nodes/chunk_append/planner.c

```diff
@@ -226,15 +231,31 @@ ts_chunk_append_plan_create(PlannerInfo *root, RelOptInfo *rel, CustomPath *path
 				merge_plan->sortOperators = sortOperators;
 				merge_plan->collations = collations;
 				merge_plan->nullsFirst = nullsFirst;
+				bool partial_chunks = false;
+
+				/* to determine if we have partial chunks */
+				if (list_length(merge_path->subpaths) == 2)
+				{
+					Path *child1 = (Path *) linitial(merge_path->subpaths);
+					Path *child2 = (Path *) lsecond(merge_path->subpaths);
+					if (child1->parent->relid == child2->parent->relid)
```

Comment:
Could we detect the partially compressed chunk in another way?

I was able to trigger [an assert](https://github.com/postgres/postgres/blob/e7bff46e50b85c6b37723426188498dab973806c/src/backend/partitioning/partprune.c#L295) in the PostgreSQL code with the current way of having the same relid for both subpaths. So, I have some plans to change the relid soon (i.e., use the relid of the compressed and the uncompressed relation) .

Reply:
Maybe I misunderstand what you are saying but the tuples produced by the node need to match the relation so it this stage there should never be any references to uncompressed tuples as they have different layout and cannot mix in an append node

Reply:
after bringing this up at standup yesterday:
I checked what effect setting `enable_partitionwise_aggregate = on` has: the ChunkAppend is not applied and instead a MergeAppend node appears on top of the plan:
```
Custom Scan (ChunkAppend) on test_partials
   Order: test_partials."time"
   ->  Custom Scan (DecompressChunk) on _hyper_1_1_chunk
         ->  Sort
               Sort Key: compress_hyper_2_8_chunk._ts_meta_sequence_num DESC
               ->  Seq Scan on compress_hyper_2_8_chunk
   ->  Custom Scan (DecompressChunk) on _hyper_1_2_chunk
         ->  Sort
               Sort Key: compress_hyper_2_9_chunk._ts_meta_sequence_num DESC
               ->  Seq Scan on compress_hyper_2_9_chunk
   ->  Merge Append
         Sort Key: _hyper_1_3_chunk."time"
         ->  Custom Scan (DecompressChunk) on _hyper_1_3_chunk
               ->  Sort
                     Sort Key: compress_hyper_2_10_chunk._ts_meta_sequence_num DESC
                     ->  Seq Scan on compress_hyper_2_10_chunk
         ->  Sort
               Sort Key: _hyper_1_3_chunk."time"
               ->  Seq Scan on _hyper_1_3_chunk
   ->  Custom Scan (DecompressChunk) on _hyper_1_7_chunk
         ->  Sort
               Sort Key: compress_hyper_2_11_chunk._ts_meta_sequence_num DESC
               ->  Seq Scan on compress_hyper_2_11_chunk
(23 rows)
```
becomes instead
```
 Merge Append
   Sort Key: _hyper_1_1_chunk."time"
   ->  Custom Scan (DecompressChunk) on _hyper_1_1_chunk
         ->  Sort
               Sort Key: compress_hyper_2_8_chunk._ts_meta_sequence_num DESC
               ->  Seq Scan on compress_hyper_2_8_chunk
   ->  Custom Scan (DecompressChunk) on _hyper_1_2_chunk
         ->  Sort
               Sort Key: compress_hyper_2_9_chunk._ts_meta_sequence_num DESC
               ->  Seq Scan on compress_hyper_2_9_chunk
   ->  Custom Scan (DecompressChunk) on _hyper_1_3_chunk
         ->  Sort
               Sort Key: compress_hyper_2_10_chunk._ts_meta_sequence_num DESC
               ->  Seq Scan on compress_hyper_2_10_chunk
   ->  Sort
         Sort Key: _hyper_1_3_chunk."time"
         ->  Seq Scan on _hyper_1_3_chunk
   ->  Custom Scan (DecompressChunk) on _hyper_1_7_chunk
         ->  Sort
               Sort Key: compress_hyper_2_11_chunk._ts_meta_sequence_num DESC
               ->  Seq Scan on compress_hyper_2_11_chunk
(21 rows)
```
but no assertion is hit. 

As discussed, I will not attempt another approach in this PR.